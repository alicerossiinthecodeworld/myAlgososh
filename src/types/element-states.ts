export enum ElementStates {
  Default = "default",
  Changing = "changing",
  Modified = "modified",
}

export const colorMap = new Map();
colorMap.set(ElementStates.Default, '4px solid rgb(0, 50, 255)')
colorMap.set(ElementStates.Changing, '4px solid rgb(210, 82, 225)')
colorMap.set(ElementStates.Modified, '4px solid rgb(127, 224, 81)')
