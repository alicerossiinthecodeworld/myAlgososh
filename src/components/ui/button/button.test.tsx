import { render, screen, fireEvent } from '@testing-library/react';
import {Button} from './button'
import { Direction } from '../../../types/direction';


const test_on_click =(isLoader:boolean,isDisabled:boolean, testid:string, callTimes:number)=> {
  const onClickMock = jest.fn();
  render(
    <Button
      data-testid="loading-button"
      text="Click me!"
      onClick={onClickMock}
      isLoader={isLoader}
      disabled={isDisabled}
      sorting={Direction.Ascending}
    />
  );
  fireEvent.click(screen.getByTestId('loading-button'));
  expect(onClickMock).toHaveBeenCalledTimes(callTimes);
}

describe('Button Component', () => {
  test('renders button with text', () => {
    const { asFragment } = render(<Button>'Click me!'</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders button without text', () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders disabled button', () => {
    const { asFragment } = render(<Button disabled={true}>Disabled</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders loading button', () => {
    const { asFragment } = render(<Button isLoader={true}>Loading</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('Button Component', () => {
    test('calls onClick callback when clicked and is not loading or disabled', () => {
      test_on_click(false, false, 'clickable-button', 1)
    });
  });

  describe('Button Component', () => {
    test('does not call on when loading', () => {
      test_on_click(true, false, 'loading-button', 0)
    });
    
  });
  describe('Button Component', () => {
    test('does not call on when disabled', () => {
      test_on_click(false, true, 'disabled-button', 0)
    });
  });
});
