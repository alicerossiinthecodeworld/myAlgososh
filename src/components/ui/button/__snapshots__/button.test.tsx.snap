// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders button with text 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`Button Component renders button without text 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`Button Component renders disabled button 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined false "
    disabled=""
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</DocumentFragment>
`;

exports[`Button Component renders loading button 1`] = `
<DocumentFragment>
  <button
    class="text text_type_button text_color_primary button undefined loader "
    disabled=""
    type="button"
  >
    <img
      alt="Загрузка."
      class="loader_icon"
      src="loader.svg"
    />
  </button>
</DocumentFragment>
`;
